src/test/java/com/example/librarymanagement/data/TestDataFactory.kt

```diff
@@ -0,0 +1,191 @@
+package com.example.librarymanagement.data
+
+import com.example.librarymanagement.dto.BookRequestDto
+import com.example.librarymanagement.dto.UserRequestDto
+import com.example.librarymanagement.model.entity.Author
+import com.example.librarymanagement.model.entity.Book
+import com.example.librarymanagement.model.entity.BookPresence
+import com.example.librarymanagement.model.entity.Journal
+import com.example.librarymanagement.model.entity.Library
+import com.example.librarymanagement.model.entity.Publisher
+import com.example.librarymanagement.model.entity.Reservation
+import com.example.librarymanagement.model.entity.User
+import com.example.librarymanagement.model.enums.Availability
+import com.example.librarymanagement.model.enums.Genre
+import java.time.LocalDate
+
+object TestDataFactory {
+
+    fun createLibrary(): Library = Library(1L, "Library", "Address")
+
+    fun createPublisher(): Publisher = Publisher(1L, "Publisher")
+
+    fun createAuthor(): Author = Author(1L, "Author", "Author")
+
+    fun createBook(): Book {
+        val author = createAuthor()
+        val publisher = createPublisher()
+        return Book(
+            id = 1L,
+            title = "Book",
+            author = author,
+            publisher = publisher,
+            publishedYear = 2021,
+            genre = Genre.DRAMA,
+            isbn = 1234567890123)
+    }
+
+    fun createUser(): User {
+        return User(
+            id = 1L,
+            firstName = "First",
+            lastName = "First",
+            email = "first@email.com",
+            phoneNumber = "1234567890",
+            password = "Password1"
+        )
+    }
+
+    fun createBookPresence(): BookPresence {
+        val book = createBook()
+        val library = createLibrary()
+        val user = createUser()
+        return BookPresence(
+            id = 1L,
+            availability = Availability.AVAILABLE,
+            book = book,
+            library = library,
+            user = user
+        )
+    }
+
+    fun createJournal(user: User, bookPresence: BookPresence): Journal {
+        return Journal(
+            id = 1L,
+            user = user,
+            bookPresence = bookPresence,
+            dateOfBorrowing = LocalDate.now(),
+            dateOfReturning = LocalDate.now().plusDays(7)
+        )
+    }
+
+    fun createJournalWithoutUser(): Journal {
```

Comment:
I'd suggest you to refactor test data factory it to the following structure
```
fun createJournal() = Journal(
            id = 1L,
            dateOfBorrowing = LocalDate.now(),
            dateOfReturning = LocalDate.now().plusDays(7)
)

fun Journal.withUser(user: User = createUser()) = copy(user = user) 
// OR if it doest not have copy function then it could be 
= Journal(
    id,
     user,
      bookPresence,
      dateOfBorrowing,
      dateOfReturning,
      )
```

But test data which is not related to the author could be remove from current PR.